### DISCUSSION TITLE
Registering on_disconnect event

### DISCUSSION THREAD

**Discussion by rbevin777** [ThomasGerstenberg/blatann#141] [General]
Hi, I'm using this library at the moment on a project. It's a very good library. I'm having a problem though, when trying to register my on `on_disconnect` callback function. It doesn't seem to register my callback. According to the documentation this will happen if there is a callback already registered. I have had a look through the code and this callback is already registered to another event within the `blatann/gatt/managers.py` file. I believe this is a bug because I cannot register any event with the `on_disconnect` method. Would you be able to confirm this if this is expected behaviour? Thanks

**Comment by ThomasGerstenberg** [comment]
You should be able to register an event handler to the `on_disconnect` event, see the example [here](https://github.com/ThomasGerstenberg/blatann/blob/master/blatann/examples/peripheral.py#L295).

Could you post a minimal code snippet that has the issue?

**Reply by rbevin777** [reply]
Hi, thanks for the response. Yea we managed to get this working. the `on_disconnect` was routed through the following function
```
_peer = peer.Client()
```
Which didn't work for some reason. But when I used `ble_device._peer.on_disconnect.register(cbk_func)`  this worked. I'm still unsure why the change worked. When I fully understand it I'll update this comment further in case someone else has an issue.
